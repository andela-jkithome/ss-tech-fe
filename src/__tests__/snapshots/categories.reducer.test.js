import reducer from '../../reducers/categories';

const initialState = {
  fetching: false,
  categories: [],
  error: null,
};

const fetchedCategories = [
  'explicit',
  'dev',
  'movie',
  'food',
  'celebrity',
  'science',
  'sport',
  'political',
  'religion',
  'animal',
  'history',
  'music',
  'travel',
  'career',
  'money',
  'fashion',
];

describe('categories reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CATEGORIES_REQUEST', () => {
    expect(
      reducer(initialState, {
        type: 'CATEGORIES_REQUEST',
      }),
    ).toMatchSnapshot();
  });

  it('should handle CATEGORIES_SUCCESS', () => {
    expect(
      reducer(Object.assign({}, initialState, { fetching: true }), {
        type: 'CATEGORIES_SUCCESS',
        categories: fetchedCategories,
      }),
    ).toMatchSnapshot();
  });

  it('should handle CATEGORIES_FAILURE', () => {
    expect(
      reducer(initialState, {
        type: 'CATEGORIES_FAILURE',
        failure: 'Error Fetching Categories!',
      }),
    ).toMatchSnapshot();
  });
});
