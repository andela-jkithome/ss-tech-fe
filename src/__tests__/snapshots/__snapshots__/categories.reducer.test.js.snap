// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories reducer should handle CATEGORIES_FAILURE 1`] = `
Object {
  "categories": Array [],
  "error": "Error Fetching Categories!",
  "fetching": false,
}
`;

exports[`categories reducer should handle CATEGORIES_REQUEST 1`] = `
Object {
  "categories": Array [],
  "error": null,
  "fetching": true,
}
`;

exports[`categories reducer should handle CATEGORIES_SUCCESS 1`] = `
Object {
  "categories": Array [
    "explicit",
    "dev",
    "movie",
    "food",
    "celebrity",
    "science",
    "sport",
    "political",
    "religion",
    "animal",
    "history",
    "music",
    "travel",
    "career",
    "money",
    "fashion",
  ],
  "error": null,
  "fetching": false,
}
`;

exports[`categories reducer should return the initial state 1`] = `
Object {
  "categories": Array [],
  "error": null,
  "fetching": false,
}
`;
